rpc_jobs/leapfrog_multi_node_aio.yml

```diff
@@ -0,0 +1,136 @@
+          FLAVOR: "onmetal-io1"
+          REGION: "IAD"
+      - rpc_repo_params:
+          RPC_BRANCH: "{branch}"
+      - osa_ops_params:
+          OPENSTACK_ANSIBLE_BRANCH: "{OPENSTACK_ANSIBLE_BRANCH}"
+          DEFAULT_IMAGE: "3.13.0-98"
+          DATA_DISK_DEVICE: "sdb"
+      - string:
+          name: STAGES
+          default: |
+            Allocate Resources, Connect Slave, Prepare Multi-Node AIO,
+             Prepare RPC Configs, Deploy RPC w/ Script, Prepare MaaS,
+             Setup MaaS, Verify MaaS, Leapfrog Upgrade
+             Install Tempest, Tempest Tests, Holland, Cleanup,
+             Destroy Slave
+          description: |
+            Pipeline stages to run CSV. Note that this list does not influence execution order.
+            Options:
+              Allocate Resources
+              Connect Slave
+              Prepare Multi-Node AIO
+              Prepare RPC Configs
+              Deploy RPC w/ Script
+              Prepare MaaS
+              Setup MaaS
+              Verify MaaS
+              Leapfrog Upgrade
+              Install Tempest
+              Tempest Tests
+              Holland
+              Pause (use to hold instance for investigation before cleanup)
+              Cleanup
+              Destroy Slave
+
+    dsl: |
+      node(){{
+        try {{
+          dir("rpc-gating"){{
+            git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
+            pubCloudSlave = load 'pipeline_steps/pubcloud.groovy'
+            common = load 'pipeline_steps/common.groovy'
+            multi_node_aio_prepare = load 'pipeline_steps/multi_node_aio_prepare.groovy'
+            deploy = load 'pipeline_steps/deploy-kilo.groovy'
+            maas = load 'pipeline_steps/maas.groovy'
+            tempest = load 'pipeline_steps/tempest.groovy'
+            horizon = load 'pipeline_steps/horizon.groovy'
+            kibana = load 'pipeline_steps/kibana.groovy'
+            holland = load 'pipeline_steps/holland.groovy'
+            github = load 'pipeline_steps/github.groovy'
+          }}
+          instance_name = common.gen_instance_name()
+          pubCloudSlave.getPubCloudSlave(instance_name: instance_name)
+
+          node(instance_name){{
+            multi_node_aio_prepare.prepare()
+            deploy.deploy_sh(
+              vm: "deploy1",
+              environment_vars: [
+                "DEPLOY_HAPROXY=yes",
+                "DEPLOY_ELK=yes",
+                "DEPLOY_TEMPEST=no",
+                "DEPLOY_AIO=no",
+                "DEPLOY_MAAS=no"
+                ]
+            ) // deploy_sh
+            multi_node_aio_prepare.multi_node_aio_networking()
+            dir("rpc-gating"){{
+              // Checkout rpc-gating repo to avoid race conditions in parallel steps
+              git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
+            }}
+            deploy.upgrade_leapfrog(vm: "deploy1", environment_vars: environment_vars)
```

Comment:
You'll want to run the holland/maas/kibana/tempest tests before the upgrade as well as after it.